pandas/_testing/asserters.py

```diff
@@ -1443,3 +1445,9 @@ def assert_metadata_equivalent(
             assert val is None
         else:
             assert val == getattr(right, attr, None)
+
+
+def assert_maybe_raises(condition, exception, match):
+    if condition:
+        return pytest.raises(exception, match=match)
+    return contextlib.nullcontext()
```

Comment:
this was just a temporary thing i was trying, definitely not suggesting it as part of the api - cleaning it up in the next commit 😇